tests/third_party/proxy/index.ts

```diff
@@ -465,4 +466,23 @@ function requestAuthorization(
 	};
 	res.writeHead(407, headers);
 	res.end('Proxy authorization required');
+}
+
+function onupgrade(req: http.IncomingMessage, socket: net.Socket, head: Buffer) {
```

Comment:
This feels like the right place because it's part of the proxy implementation, but also like the wrong place because it's in a vendored file. Let me know if I should move this into `tests/config/proxy.ts` instead.